private/engine/config.go

```diff
@@ -1,52 +1,121 @@
+	ClientCert      string // path to file containing client certificate (or chain of certs)
+	ClientKey       string // path to file containing client's private key
+	ClientTimeout   uint   // timeout for overall client call (seconds), zero means timeout disabled
+	IdleConnTimeout uint   // timeout for idle connection (seconds), zero means timeout disabled
+	WriteBufferSize int    // size of the write buffer (bytes), if zero then uses http.Transport default
+	ReadBufferSize  int    // size of the read buffer (bytes), if zero then uses http.Transport default
 }
 
-var DefaultConfig = Config{
+var DefaultSocketTimeouts = socketConfig{
 	DialTimeout:           1,
 	RequestTimeout:        5,
 	ResponseHeaderTimeout: 5,
 }
+var DefaultHttpTimeouts = httpConfig{
+	ClientTimeout:   10,
+	IdleConnTimeout: 10,
+}
 
-// LoadConfig sets up the configuration for the connection to a txn manager.
+func IsSocketConfigured(cfg Config) bool {
+	return cfg.ConnectionType == socketConnection
+}
+
+func IsTlsConfigured(cfg Config) bool {
+	return cfg.HttpConfig.Tls == "STRICT"
+}
+
+// FetchConfig sets up the configuration for the connection to a txn manager.
 // It will accept a path to a socket file or a path to a config file,
-// and returns the full configuration info for the socket file.
-func LoadConfig(path string) (Config, error) {
+// and returns the full configuration info for either a socket file or HTTP.
+func FetchConfig(path string) (Config, error) {
 	info, err := os.Lstat(path)
 	if err != nil {
 		return Config{}, err
 	}
 
-	cfg := DefaultConfig
+	var cfg Config
 	isSocket := info.Mode()&os.ModeSocket != 0
-	if !isSocket {
-		if _, err := toml.DecodeFile(path, &cfg); err != nil {
+	if isSocket {
+		cfg.ConnectionType = socketConnection
+		cfg.SocketConfig = DefaultSocketTimeouts
+		cfg.SocketConfig.WorkDir, cfg.SocketConfig.Socket = filepath.Split(path)
+
+	} else {
+		err = LoadConfigFile(path, &cfg)
+		if err != nil {
 			return Config{}, err
 		}
-	} else {
-		cfg.WorkDir, cfg.Socket = filepath.Split(path)
 	}
 
-	// Fall back to Constellation 0.0.1 config format if necessary
-	if cfg.Socket == "" {
-		cfg.Socket = cfg.SocketPath
-	}
 	return cfg, nil
 }
+
+func LoadConfigFile(path string, cfg *Config) error {
+	cfg.SocketConfig = DefaultSocketTimeouts
+	cfg.HttpConfig = DefaultHttpTimeouts
+
+	if _, err := toml.DecodeFile(path, cfg); err != nil {
+		return err
+	}
+	cfg.HttpConfig.Tls = strings.ToUpper(cfg.HttpConfig.Tls)
+
+	if cfg.SocketConfig.Socket != "" && cfg.HttpConfig.Url != "" {
+		return fmt.Errorf("cannot specify both Socket and HTTP connections in config file")
+	}
+
+	if cfg.SocketConfig.Socket != "" {
+		cfg.ConnectionType = socketConnection
+	} else if cfg.HttpConfig.Url != "" {
+		cfg.ConnectionType = httpConnection
+		switch cfg.HttpConfig.Tls {
+		case "OFF":
+			//no action needed
+		case "STRICT":
+			if cfg.HttpConfig.RootCA == "" || cfg.HttpConfig.ClientCert == "" || cfg.HttpConfig.ClientKey == "" {
```

Comment:
I think we are missing some options here
- RootCA can be empty in which case Go defaults to host certificates
- ClientCert & ClientKey can be empty and opening a TLS connexion is still possible. Usually, ClientCert & ClientKey are used when required by a server for mutual authentication

Reply:
First point is done.
Second point is not done as Tessera does not support 1-way TLS. (It could be added if Tessera is changed to implement this in future).